src/axom/sina/tests/sina_Document.cpp

```diff
@@ -447,6 +565,276 @@ TEST(Document, load_defaultRecordLoaders)
+  actual = node_to_double_vector(rec0["curve_sets"]["1"]["dependent"]["1"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {4, 5, 4.0, 5.0};
+  actual = node_to_double_vector(rec0["curve_sets"]["1"]["independent"]["0"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {7, 8, 7.0, 9.0};
+  actual = node_to_double_vector(rec0["curve_sets"]["1"]["independent"]["1"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {1.0, 1.0};
+  actual = node_to_double_vector(rec0["curve_sets"]["2"]["dependent"]["0"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {10.0, 10.0};
+  actual = node_to_double_vector(rec0["curve_sets"]["2"]["dependent"]["1"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {4.0, 4.0};
+  actual = node_to_double_vector(rec0["curve_sets"]["2"]["independent"]["0"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {7.0, 7.0};
+  actual = node_to_double_vector(rec0["curve_sets"]["2"]["independent"]["1"]["value"]);
+  ASSERT_EQ(actual, expected);
+
+  // ----- Record 1 -----
+  const conduit::Node &rec1 = root["records"].child(1);
+  ASSERT_EQ(rec1["id"].as_string(), "bar2");
+  ASSERT_EQ(rec1["type"].as_string(), "foo2");
+
+  expected = {1, 2};
+  actual = node_to_double_vector(rec1["curve_sets"]["1"]["dependent"]["0"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {10, 20};
+  actual = node_to_double_vector(rec1["curve_sets"]["1"]["dependent"]["1"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {4, 5};
+  actual = node_to_double_vector(rec1["curve_sets"]["1"]["independent"]["0"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {7, 8};
+  actual = node_to_double_vector(rec1["curve_sets"]["1"]["independent"]["1"]["value"]);
+  ASSERT_EQ(actual, expected);
+
+  // ----- Record 2 -----
+  const conduit::Node &rec2 = root["records"].child(2);
+  ASSERT_EQ(rec2["id"].as_string(), "bar3");
+  ASSERT_EQ(rec2["type"].as_string(), "foo2");
+
+  expected = {1.0, 1.0};
+  actual = node_to_double_vector(rec2["curve_sets"]["1"]["dependent"]["0"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {10.0, 10.0};
+  actual = node_to_double_vector(rec2["curve_sets"]["1"]["dependent"]["1"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {4.0, 4.0};
+  actual = node_to_double_vector(rec2["curve_sets"]["1"]["independent"]["0"]["value"]);
+  ASSERT_EQ(actual, expected);
+  expected = {7.0, 7.0};
+  actual = node_to_double_vector(rec2["curve_sets"]["1"]["independent"]["1"]["value"]);
+  ASSERT_EQ(actual, expected);
+}
+
+TEST(Document, test_append_to_json_failures)
+{
+  // We have a lot of potential failures
+  // 1. Fail validation due to an appended-to curve not being uniform
+  // 2. Fail validation due to a new curve not being uniform
+  // 3. Fail because Dependents won't be uniform with Independents
+  // 4. Fail because Protocol 3 Duplicate Data
+  // 5. Fail because Protocol 3 Duplicate UDC
+
+  std::streambuf *origBuffer = std::cerr.rdbuf();
```

Comment:
If tests are running in parallel, is there a guarantee that this test will have exclusive access to `cerr`?

Is there a way these tests can be rewritten that leverages the return values rather than the output to cerr?

Reply:
A typical conduit pattern is to return the error in an optional conduit node.